homeassistant/components/home_plus_control/config_flow.py

```diff
@@ -0,0 +1,161 @@
+"""Config flow for Legrand Home+ Control."""
+import logging
+import re
+import time
+from typing import Any, Dict, Optional
+
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_CLIENT_ID, CONF_CLIENT_SECRET
+from homeassistant.core import callback
+from homeassistant.helpers import config_entry_oauth2_flow, config_validation as cv
+
+from .const import (
+    CONF_MODULE_STATUS_UPDATE_INTERVAL,
+    CONF_PLANT_TOPOLOGY_UPDATE_INTERVAL,
+    CONF_PLANT_UPDATE_INTERVAL,
+    CONF_SUBSCRIPTION_KEY,
+    DEFAULT_UPDATE_INTERVALS,
+    DOMAIN,
+)
+
+
+class HomePlusControlFlowHandler(
+    config_entry_oauth2_flow.AbstractOAuth2FlowHandler, domain=DOMAIN
+):
+    """Config flow to handle Home+ Control OAuth2 authentication."""
+
+    DOMAIN = DOMAIN
+
+    # Pick the Cloud Poll class
+    CONNECTION_CLASS = config_entries.CONN_CLASS_CLOUD_POLL
+
+    @property
+    def logger(self) -> logging.Logger:
+        """Return logger."""
+        return logging.getLogger(__name__)
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow start initiated by the user."""
+        await self.async_set_unique_id(DOMAIN)
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        return await super().async_step_user(user_input)
+
+    async def async_step_creation(
```

Comment:
Override `async_oauth_create_entry` instead if needed. But I don't think we need to do that either. We probably don't need to store client id, secret or subscription key in entry dat since that's available on the implementation.

```py
async def async_oauth_create_entry(self, data: dict) -> dict:
    """Create an entry for the flow.

    Ok to override if you want to fetch extra info or even add another step.
    """
    return self.async_create_entry(title=self.flow_impl.name, data=data)
```